arkouda/array_view.py

```diff
@@ -0,0 +1,249 @@
+from arkouda.pdarrayclass import pdarray, parse_single_value, create_pdarray
+from arkouda.pdarraycreation import array, arange, zeros
+from arkouda.numeric import cumprod, where, cast as akcast
+import numpy as np  # type: ignore
+from arkouda.client import generic_msg
+from arkouda.dtypes import translate_np_dtype, resolve_scalar_dtype
+from enum import Enum
+import json
+
+OrderType = Enum('OrderType', ['ROW_MAJOR', 'COLUMN_MAJOR'])
+
+
+class ArrayView:
+    """
+    A multi-dimensional view of a pdarray. Arkouda ``ArraryView`` behaves similarly to numpy's ndarray.
+    The base pdarray is stored in 1-dimension but can be indexed and treated logically as if it were multi-dimensional
+
+    Attributes
+    ----------
+    base: pdarray
+        The base pdarray that is being viewed as a multi-dimensional object
+    dtype: dtype
+        The element type of the base pdarray (equivalent to base.dtype)
+    size: int_scalars
+        The number of elements in the base pdarray (equivalent to base.size)
+    shape: pdarray[int]
+        A pdarray specifying the sizes of each dimension of the array
+    ndim: int_scalars
+         Number of dimensions (equivalent to shape.size)
+    itemsize: int_scalars
+        The size in bytes of each element (equivalent to base.itemsize)
+    order: str {'C'/'row_major' | 'F'/'column_major'}
+        Index order to read and write the elements.
+        By default or if 'C'/'row_major', read and write data in row_major order
+        If 'F'/'column_major', read and write data in column_major order
+    """
+    def __init__(self, base: pdarray, shape, order='row_major'):
+        self.objtype = type(self).__name__
+        self.shape = array(shape)
+        if not isinstance(self.shape, pdarray):
+            raise TypeError(f"Shape cannot be type {type(self.shape)}")
```

Comment:
Yeah I can update this. I don't actually expect this to ever be rasied, I think i mostly did this for typechecking reasons since I want to treat shape as a pdarray and `ak.array` can return a `Strings`